compositor_common/src/error.rs

```diff
@@ -44,6 +44,23 @@ impl Display for UnusedNodesError {
     }
 }
 
+#[derive(Debug, thiserror::Error, PartialEq, Eq)]
+pub enum UnsatisfiedConstraintsError {
+    #[error(
+        "Invalid input pads specification for node \"{}\". {} requires {}. {}",
+        node_id,
+        identification_name,
+        input_count_constrain.required_inputs_message(),
+        Self::defined_inputs_message(defined_input_pads_count)
+    )]
+    InvalidInputsCount {
```

Comment:
consider sth like this

```rust
pub enum UnsatisfiedConstraintsError {
  #[error(transparent)]
  InvalidInputCount(#[from] InvalidInputCountError)
}

#[derive(*)]
struct InputCountValidationError {
        identification_name: Rc<str>,
        input_count_constrain: InputsCountConstraint,
        defined_input_pads_count: u32,
}

impl Display for InputCountValidationError {

}
```

and try to avoid functions that return some string that you will embed in error message